src/accommodation/accommodation.service.ts

```diff
@@ -153,4 +154,76 @@ export class AccommodationService {
     }
     return accommodation;
   }
+
+  async getAllAccommodations(options: OrderAndFilter) {
+    try {
+      const findManyOptions = this.generateFindAllQueryObj(options);
+
+      const [accommodations, totalCount, priceStats] = await Promise.all([
+        this.prisma.accommodation.findMany(findManyOptions),
+        this.prisma.accommodation.count({
+          where: findManyOptions.where,
+        }),
+        this.prisma.accommodation.aggregate({
+          _min: { price: true },
+          _max: { price: true },
+          where: findManyOptions.where,
+        }),
+      ]);
+
+      const {
+        _min: { price: minPrice },
+        _max: { price: maxPrice },
+      } = priceStats;
+
+      return { priceRange: { minPrice, maxPrice }, totalCount, data: accommodations };
+    } catch (error) {
+      throw new GlobalException(ErrorsTypes.ACCOMMODATIONS_LIST_FAILED_TO_GET, error.message);
+    }
+  }
+
+  private generateFindAllQueryObj(options: OrderAndFilter) {
+    const findManyOptions: any = {
+      select: {
+        id: true,
+        thumbnailUrl: true,
+        squareMeters: true,
+        numberOfRooms: true,
+        allowedNumberOfPeople: true,
+        price: true,
+        address: {
+          select: {
+            country: true,
+          },
+        },
+      },
+
+      where: {
+        availability: true,
+        price: {
+          gte: options.minPrice,
+          lte: options.maxPrice,
+        },
+        numberOfRooms: {
+          gte: options.minRooms,
+          lte: options.maxRooms,
+        },
+        allowedNumberOfPeople: {
+          gte: options.minPeople,
+          lte: options.maxPeople,
+        },
+      },
+
+      skip: (options.page! - 1) * options.limit!,
+      take: options.limit,
+    };
+
+    if (options.orderBy) {
+      findManyOptions.orderBy = {
```

Comment:
As I remember there are several fields to order simultaniously

Reply:
done